tests/server/auth/test_sqlalchemy_store.py

```diff
@@ -0,0 +1,135 @@
+    _AUTH_CONFIG_PATH_ENV_VAR,
+)
+from mlflow.server.auth.config import read_auth_config
+from mlflow.server.auth.entities import User
+from mlflow.server.auth.sqlalchemy_store import (
+    SqlUser,
+    SqlExperimentPermission,
+    SqlRegisteredModelPermission,
+    SqlAlchemyStore,
+)
+from mlflow.protos.databricks_pb2 import (
+    ErrorCode,
+    RESOURCE_DOES_NOT_EXIST,
+    INVALID_PARAMETER_VALUE,
+    RESOURCE_ALREADY_EXISTS,
+)
+from tests.helper_functions import random_str
+
+pytestmark = pytest.mark.notrackingurimock
+
+
+def _get_db_uri_from_env_var():
+    auth_config_path = os.environ.get(_AUTH_CONFIG_PATH_ENV_VAR)
+    if auth_config_path:
+        auth_config = read_auth_config(auth_config_path)
+        return auth_config.database_uri
+
+
+@pytest.fixture
+def store(tmp_sqlite_uri):
+    db_uri_from_env_var = _get_db_uri_from_env_var()
+    store = SqlAlchemyStore()
+    store.init_db(db_uri_from_env_var if db_uri_from_env_var else tmp_sqlite_uri)
+    yield store
+
+    if db_uri_from_env_var is not None:
+        with store.ManagedSessionMaker() as session:
+            for model in (
+                SqlUser,
+                SqlRegisteredModelPermission,
+                SqlExperimentPermission,
+            ):
+                session.query(model).delete()
+
+
+def _user_maker(store, username, password, is_admin=False):
+    return store.create_user(username, password, is_admin)
+
+
+def test_create_user(store):
+    username1 = random_str()
+    password1 = random_str()
+    user1 = _user_maker(store, username1, password1)
+    assert user1.username == username1
+    assert user1.password_hash != password1
+    assert user1.is_admin is False
+
+    # error on duplicate
+    with pytest.raises(
+        MlflowException, match=rf"User \(username={username1}\) already exists"
+    ) as exception_context:
+        _user_maker(store, username1, password1)
+    assert exception_context.value.error_code == ErrorCode.Name(RESOURCE_ALREADY_EXISTS)
+
+    # slightly different name is ok
+    username2 = username1 + "_2"
+    password2 = password1 + "_2"
+    user2 = _user_maker(store, username2, password2, is_admin=True)
+    assert user2.username == username2
+    assert user2.password_hash != password2
+    assert user2.is_admin is True
+
+    # invalid username will fail
+    with pytest.raises(MlflowException, match=r"Username cannot be empty") as exception_context:
+        _user_maker(store, None, None)
+    assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+    with pytest.raises(MlflowException, match=r"Username cannot be empty") as exception_context:
+        _user_maker(store, "", "")
+    assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+
+
+def test_has_user(store):
+    username1 = random_str()
+    password1 = random_str()
+    _user_maker(store, username1, password1)
+    assert store.has_user(username=username1) is True
```

Comment:
```suggestion
    assert store.has_user(username=username1)
```

Do we need `is True` here?

Reply:
yes, i think it's better to be more explicit in testing